database/buildSchema/35_file.sql

```diff
@@ -36,3 +38,26 @@ CREATE TRIGGER file_deletion
     FOR EACH ROW
     EXECUTE FUNCTION public.notify_file_server ();
 
+-- FUNCTION to mark file for deletion if it's no longer a reference doc
+CREATE OR REPLACE FUNCTION public.mark_file_for_deletion ()
+    RETURNS TRIGGER
+    AS $file_event$
+BEGIN
+    UPDATE
+        public.file
+    SET
+        to_be_deleted = TRUE
+    WHERE
+        id = NEW.id;
+    RETURN NULL;
+END;
+$file_event$
+LANGUAGE plpgsql;
+
+-- TRIGGER to execute above function when files no longer reference
+CREATE TRIGGER file_no_longer_reference
```

Comment:
Hmm, as I was reading this, I was contemplating the complexity repercussions of indirection, and was wondering if it's easier to follow if this was a seperate action, instead of using modifyRecord. We could also reduce complexity of template configuration (even though it would require double upload if document is shared between internal and external, i think general UI would be simpler to use and configurations would also be simpler)

i.e. 

## Template

Just two file upload form elements, one for internal, and another for external (with default values from graphql file request)

## Action 

updateReferenceDocuments
two parameters, for codes of form elements for internal and external file.

Action internally insert/updates or deletes relevant file entries.